sos/report/plugins/kdump.py

```diff
@@ -102,4 +103,25 @@ def setup(self):
             "/etc/default/kdump-tools"
         ])
 
+
+class CosKDump(KDump, CosPlugin):
+
+    option_list = [
+        ("all_dumps", "enable capture for all kernel dumps", "", False),
+        ("latest_dump", "enable capture for latest kernel crash dump",
+         "", False),
+    ]
+
+    def setup(self):
+        super(CosKDump, self).setup()
+        self.add_cmd_output('ls -alRh /var/kdump*')
+        if self.get_option("all_dumps"):
+            self.add_copy_spec([
+                "/var/kdump-*"
+            ])
+        if self.get_option("latest_dump"):
+            self.add_copy_spec([
+                "/var/kdump"
+            ])
```

Comment:
Collections like this should leverage the `--all-logs` global option instead of using plugin-specific options. E.G.

```python
if self.get_option('all_logs'):
    self.add_copy_spec('/var/kdump-*')
else:
    self.add_copy_spec('/var/kdump')
```

The above would result in always _trying_ to capture a kdump file, but may be prevented if that kdump file is larger than the sizelimit (25MB by default). The behavior of `add_copy_spec()` will ensure that the newest file is always taken first. If you want, you can also pass `sizelimit=XYZ` to the `add_copy_spec()` to increase the threshold for capturing a kdump file or not. Since this is limited to the CoS policy variant of the plugin, feel free to set this as your org needs.

Reply:
Thank you for the suggestion but we would want to use the all_dumps to collect the kdump file rather than all_logs.

Reply:
Plugin options are not meant to duplicate global options.

If you want this collection to be option-gated, then you can use a single option to do the gating and rely on the existing behavior of `--all-logs` and `add_copy_spec()`:

```python
PluginOpt('collect-kdumps', default=False, desc='Collect existing kdump files')
[...]
if self.get_option('collect-kdumps'):
    self.add_copy_spec('/var/kdump-*')
```

The above would capture capture files matching `/var/kdump-*` starting with the most recent kdump file, up to `sizelimit`. When `--all-logs` is used, this means _all_ kdump files will be collected.

Having two separate plugin options to dictate a single copy specification is not a user experience we want to encourage in plugins.